plainly-plugin/src/ui/components/form/UploadForm.tsx

```diff
@@ -0,0 +1,170 @@
+import FormData from 'form-data';
+import { useState } from 'react';
+import { makeProjectZip, makeTmpDir, removeFolder } from '../../../node';
+import { postFormData } from '../../../node/request';
+import Button from '../common/Button';
+import Description from '../typography/Description';
+import Label from '../typography/Label';
+import PageHeading from '../typography/PageHeading';
+
+import fs from 'fs';
+import { LoaderCircleIcon } from 'lucide-react';
+import { TMP_DIR } from '../../../node/constants';
+import { useNotification } from '../../hooks/useNotification';
+import { useSettings } from '../../hooks/useSettings';
+import Alert from '../common/Alert';
+import Notification from '../common/Notification';
+
+export default function UploadForm() {
+  const [inputs, setInputs] = useState<{
+    projectName?: string;
+    description?: string;
+    tags?: string[];
+  }>({});
+  const [loading, setLoading] = useState(false);
+  const { notification, notifySuccess, notifyError, clearNotification } =
+    useNotification();
+  const { getSettingsApiKey, loading: settingsLoading } = useSettings();
+
+  const apiKey = getSettingsApiKey();
+
+  const disabled = loading || !apiKey?.key;
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setLoading(true);
+    let collectFilesDirValue: string | undefined;
+    let zipPathValue: string | undefined;
+
+    try {
+      await makeTmpDir();
```

Comment:
* not here, again why would this be here? create `makeProjectZip()` that creates a zip in tmp folder..
* make tmp dir removed
```
makeProjectZip() {
   await fsPromises.mkdir(TMP_DIR, { recursive: true });
   await makeProjectZip(TMP_DIR);
}
```